### ISSUE TITLE
[Multi Actions] Get ready for release

### ISSUE THREAD

**Issue by fontanierh** [dust-tt/dust#5425]
# Checklist of things that need to be done before release

## No owner

 🧹  _all cards are committed or done !_

---

## Committed

- [ ] Blog Post + Linkedin
Tracked here: https://dust4ai.slack.com/archives/C05V0P20A72/p1717511258534969
Owner @Yutcam (@fontanierh to draft some bullet points)

- [ ] Video
No separate card (tracked in the current issue)
Owner @mgxlebrun 

- [ ] Track assistant builder steps to see where we lose people
Tracked here: https://github.com/dust-tt/dust/issues/5490
Owner: @lasryaric  + @mgxlebrun 

---

## Done 

- [x] Multi Action assistants should work in the Try modal
Tracked here: https://github.com/dust-tt/dust/issues/5409
Owner: @fontanierh 

- [x] Feature flag and/or command to "switch" a workspace to multi-actions
Tracked here: https://github.com/dust-tt/dust/issues/5436
Owner: @fontanierh 

- [x] Dust app run action configuration fixes
Tracked here: https://github.com/dust-tt/dust/issues/5437
Owner: @PopDaph 

- [x] Action names hidden by default
Tracked here: https://github.com/dust-tt/dust/issues/5438
Owner: @fontanierh 

- [x] Extract action configuration
Tracked here: https://github.com/dust-tt/dust/issues/5439
Owner: @fontanierh 

- [x] Investigate if / why assistant temperature is not being properly forwarded to the multi actions agent Dust app
Tracked here: https://github.com/dust-tt/dust/issues/5565
Owner: @spolu 

- [x] Display description not action name in Action card
Tracked here: https://github.com/dust-tt/dust/issues/5566
Owner: @spolu 

- [x] Library of meta-prompts (per provider / model) and save Soupinou from the JSON disease.
Tracked here: https://github.com/dust-tt/dust/issues/5548
Owner: @spolu 

- [x] Templates experience
Tracked here: https://github.com/dust-tt/dust/issues/5493
Owner: @PopDaph 

- [x] Handle "chain of thoughts" generation properly (Claude, very rarely GPT4)
Tracked here: https://github.com/dust-tt/dust/issues/5159
Owner: @fontanierh 

- [x] Browser Action
Tracked here: https://github.com/dust-tt/dust/issues/5473
Owner: @philipperolet  

- [x] Merge browser and websearch actions (+ config from builder)
Tracked here: https://github.com/dust-tt/dust/issues/5379
Owner @philipperolet 

- [x] Public-facing documentation / tutorial on Notion (+ link to it from product)
Tracked here: https://github.com/dust-tt/dust/issues/5420
Owner: @PopDaph 

- [x] Action execution "chips" need to look good
Tracked here: https://github.com/dust-tt/dust/issues/5350
Owner: @flvndvd

**Comment by lasryaric** [comment]
@fontanierh what is the scope of this task? I am not clear. Thanks.

**Comment by fontanierh** [comment]
@lasryaric Sorry I was not done writing the OP. It's now done

**Comment by fontanierh** [comment]
Regarding non-eng work, I imagine we'll want:
- a blog post + linkedin ?
- a product update email ?
- maybe a video ?

@Yutcam WDYT ?

**Comment by Yutcam** [comment]
yes! 

1. a blog post + linkedin - can you draft bullet points @fontanierh ? I can take care of the rest ;) 
2. maybe a video  - can you own it @fontanierh or @mgxlebrun maybe? I won't have the capacity, I fear. 
3. a product update email - we can draft a dedicated product update for this. Let's use the same content as in the blog. 

What about the product logs? Is there any hack you can think of to help our users discover this killer feature?

**Comment by fontanierh** [comment]
Thanks @Yutcam 🙏 , i'll draft the bullet points !

Regarding the video, i'm really bad at them, I think @mgxlebrun is a much better choice if you don't mind 🙃 

Will also try to prepare a few use-cases

**Comment by mgxlebrun** [comment]
I can own the video yes - @Yutcam let me know when you have a rough outline for the blog post so I try to create something related ?

**Comment by Yutcam** [comment]
Tks a lot @fontanierh

**Comment by fontanierh** [comment]
(i) one setup that I feel would work fine is just have it exposed as a single action with 0 config (no name no description no nothing) that is really implemented as 2 actions in the background. And just let the model decide. I've been using that 2 -actions setup extensively this past few weeks (god assistant) and it works quite well.

(ii) we can chat about this tomorrow morning. I think every action has its specifics, but IMHO we could implement "chips" (or whatever) in a much more generic way than what we have to day. At the end of the day, it's always a title, an icon, a bunch of params and some kind of output.

**Comment by spolu** [comment]
(i) yes aligned 

(ii) Let's discuss tomorrow 👍

Other questions for tomorrow:

- do we keep v1 assistant mode? What's the end state? What do we scope for a first release?

- Related: How do we productize forceAtIteration? If we have single and multi actions I presume we don't expose it, but is there another path?

**Comment by philipperolet** [comment]
Ok, since I'm owning (i) and now the browse action, I'm going to take care of this, via the following renamed issue [Websearch & Browse appear as a single action in the Builder, with no config](https://github.com/dust-tt/dust/issues/5379)

FYI @Duncid and @lasryaric since it's directly related to the discussion we had in [this thread](https://dust4ai.slack.com/archives/C050A0S2Z7F/p1716883751545989?thread_ts=1716369237.385929&cid=C050A0S2Z7F)

**Comment by fontanierh** [comment]
Summary of IRL chat: https://dust4ai.slack.com/archives/C050SM8NSPK/p1717501427441079?thread_ts=1717425835.500499&cid=C050SM8NSPK

This generated these new action items:
https://github.com/dust-tt/dust/issues/5436
https://github.com/dust-tt/dust/issues/5437
https://github.com/dust-tt/dust/issues/5438
https://github.com/dust-tt/dust/issues/5439

**Comment by mgxlebrun** [comment]
Thought: checking the new assistant creation flow, I think we will really want to understand what users do when they create assistants and where they potentially get stuck to iterate on the experience 

So I'd add a item _"set up tracking builder experience"_ towards the end of the list - it's not 100% necessary to launch and maybe having it in the "Builder experience" section [of this card about tracking](https://github.com/dust-tt/tasks/issues/789) is enough, but I'm afraid it might be dropped if not added here (and then we'll have no data like for templates)

**Comment by Duncid** [comment]
Have we checked if templates will need updating with multi action?

**Comment by fontanierh** [comment]
> So I'd add a item "set up tracking builder experience" towards the end of the list - it's not 100% necessary to launch and maybe having it in the "Builder experience" section https://github.com/dust-tt/tasks/issues/789 is enough, but I'm afraid it might be dropped if not added here (and then we'll have no data like for templates)

@mgxlebrun great point 👍  Adding an item / card for it 🙏

**Comment by fontanierh** [comment]
[Multi Actions] Track assistant builders step to figure out if / where we are losing people
[#5490](https://github.com/dust-tt/dust/issues/5490)

**Comment by mgxlebrun** [comment]
> Do you want to own framing / defining tracking plan for this ?
Happy to ! I saw the new card, I'll have a think and contribute by early next week if that works

**Comment by fontanierh** [comment]
Perfect! Thanks @mgxlebrun

**Comment by fontanierh** [comment]
@Duncid 

> Have we checked if templates will need updating with multi action?

Indeed it will be a pretty bad XP I think. Adding a card for it

**Comment by fontanierh** [comment]
New task: https://github.com/dust-tt/dust/issues/5548

**Comment by fontanierh** [comment]
New task: [[Multi Actions] Display description not action name in Action card
#5566](https://github.com/dust-tt/dust/issues/5566)

**Comment by fontanierh** [comment]
New task: [[Multi Actions] Investigate if / why assistant temperature is not being properly forwarded to the multi actions agent Dust app
#5565](https://github.com/dust-tt/dust/issues/5565)

**Comment by spolu** [comment]
This task is no longer @dust-tt/engineering-team so closing cc @mgxlebrun @Yutcam who are mentioned in it and may have left-over work.
